.github/workflows/legend-release.yml

```diff
@@ -26,9 +36,138 @@ env:
   WORKFLOW_ID: legend-stack-release.yml
 
 jobs:
+  compute-versions:
+    runs-on: ubuntu-latest
+    outputs:
+      legend_shared_release_version: ${{ steps.set_legend_shared_release_version.outputs.legend_shared_release_version }}
+      legend_pure_release_version: ${{ steps.set_legend_pure_release_version.outputs.legend_pure_release_version }}
+      legend_engine_release_version: ${{ steps.set_legend_engine_release_version.outputs.legend_engine_release_version }}
+      legend_sdlc_release_version: ${{ steps.set_legend_sdlc_release_version.outputs.legend_sdlc_release_version }}
+      legend_depot_release_version: ${{ steps.set_legend_depot_release_version.outputs.legend_depot_release_version }}
+    steps:
+      - name: Compute legend-shared-release-version
+        id: set_legend_shared_release_version
+        if: github.event.inputs.legend-shared-release != 'n/a'
+        run: |
+          GROUP_ID="org.finos.legend.shared"
+          ARTIFACT_ID="legend-shared"
+          
+          # Query Maven Central API to get latest version
+          latest_version=$(curl -s "https://search.maven.org/solrsearch/select?q=g:${GROUP_ID}+AND+a:${ARTIFACT_ID}&core=gav&rows=1&wt=json" | jq -r '.response.docs[0].v')
```

Comment:
I'm not sure querying the Maven API is the right thing to do here. There are plenty of situations where someone may attempt to run a release, create tags on the project, but then the artifact could fail to upload to Maven.

Unless they then do a clean up, this action will fail as it will look at Maven artifacts and attempt to use a tag that already exists on the project. This should be checking what is the latest tag on the GitHub repo instead.

Reply:
Thanks, done